### PR TITLE
feat(web): storage config model

### PR DESCRIPTION
**Pull request by joseivanlopez**
The UI components should be agnostic to the structure and complexity of the storage config.

Add a Config model to deal with the storage config. The model generates objects (e.g., `Drive`, `Partition`, etc) from the config data. This makes easier for the UI components to read and modify the config data.

### CODE REVIEW

File: web/src/api/storage/types/checks.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+export function isFormattedDrive(drive: config.DriveElement): drive is config.FormattedDrive {
+  return "filesystem" in drive;
+}
+
+export function isPartitionedDrive(drive: config.DriveElement): drive is config.PartitionedDrive {
+  return !("filesystem" in drive);
+}
+
+export function isSearchAll(search: config.Search): search is config.SearchAll {
```
Comment (ancorgs) [inline web/src/api/storage/types/checks.ts:35]: The name of this function can be confusing. There are many ways to search for all.

But this functions checks whether a particular way (the asterisk which is literally a shortcut for `{"search": { "ifNotFound": "skip" }}`) is used.

I would prefer a name that reflects that. I don't have a good suggestion, though. Maybe `isSearchAllShortcut` or something along that line.

File: web/src/api/storage/types/checks.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  return search === "*";
+}
+
+export function isSearchByName(search: config.Search): search is config.SearchByName {
```
Comment (ancorgs) [inline web/src/api/storage/types/checks.ts:39]: Same than above, there are several ways to search for name. This only checks if the shortcut has been used and that's not obvious from the function name.

File: web/src/api/storage/types/checks.ts (hunk 1 above)
Comment (ancorgs) [reply web/src/api/storage/types/checks.ts:35]: I know it corresponds to the name of the type `searchByName` but I still find it kind of confusing. Maybe a better name for the type, maybe a better name for this function... I don't know.

File: web/src/api/storage/types/checks.ts (hunk 2 above)
Comment (ancorgs) [reply web/src/api/storage/types/checks.ts:39]: Again same than above, I know it corresponds to the name of the type. But still I see to much room for confusion.

File: web/src/api/storage/types/checks.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  return "delete" in partition;
+}
+
+export function isPartitionToDeleteIfNeeded(
+  partition: config.PartitionElement,
+): partition is config.PartitionToDeleteIfNeeded {
+  return "deleteIfNeeded" in partition;
+}
+
+export function isPartition(partition: config.PartitionElement): partition is config.Partition {
```
Comment (ancorgs) [inline web/src/api/storage/types/checks.ts:59]: Similar to the previous cases (and again, understanding it's a consequence of the type name) this name is confusing. According to the definition, this is false for `isPartition`:

`{"search": "/dev/vda1", "size": { "min": 0, "max": "current" }, "deleteIfNeeded": true }`

While this is true for `isPartition`:

`{"search": "/dev/vda1", "size": { "min": 0, "max": "current" } }`

Sure it makes sense when you compare the three names all along each other and in the right order (`partitionToDelete`, `partitionToDeleteIfNeeded` and `partition`). But not much when you find the last one in isolation.

File: web/src/storage/model/config/size.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    const size: Size = { min: this.bytes(sizeRange.min) };
+    if (sizeRange.max) size.max = this.bytes(sizeRange.max);
+
+    return size;
+  }
+
+  private bytes(value: config.SizeValueWithCurrent): number | undefined {
+    if (checks.isSizeCurrent(value)) return;
+    // TODO: bytes from string.
+    if (checks.isSizeString(value)) return;
```
Comment (ancorgs) [inline web/src/storage/model/config/size.ts:74]: Is this omission recorded somewhere? Sounds like something we should implement very soon.

File: web/src/api/storage/types/checks.ts (hunk 1 above)
Comment (joseivanlopez) [reply web/src/api/storage/types/checks.ts:35]: Yes, it is tricky to find good names for that types. If we change it, I would change the type name. IMHO, changing the type guard name is not a good idea.

File: web/src/storage/model/config/size.ts (hunk 4 above)
Comment (joseivanlopez) [reply web/src/storage/model/config/size.ts:74]: Yes, there is a follow-up item in the trello card.

**Review by ancorgs — APPROVED** [review]
LGTM

**Pull request merged**